pkg/controller/util.go

```diff
@@ -68,19 +66,16 @@ func checkPVC(pvc *v1.PersistentVolumeClaim, annotation string) bool {
 }
 
 // returns the endpoint string which contains the full path URI of the target object to be copied.
-func getEndpoint(pvc *v1.PersistentVolumeClaim) (string, error) {
+func getEndpoint(pvc *v1.PersistentVolumeClaim) string {
 	ep, found := pvc.Annotations[AnnEndpoint]
-	if !found || ep == "" {
-		verb := "empty"
-		if !found {
-			verb = "missing"
-		}
-		return ep, errors.Errorf("annotation %q in pvc \"%s/%s\" is %s\n", AnnEndpoint, pvc.Namespace, pvc.Name, verb)
+
```

Comment:
We should keep the logging, its important to know there is no endpoint when you are expecting it. We just have to not call this function when source is none.